ModTek/IModDef.cs

```diff
@@ -19,6 +19,7 @@ public interface IModDef
 
         HashSet<string> DependsOn { get; set; }
         HashSet<string> ConflictsWith { get; set; }
+        HashSet<string> OptionalDependencies { get; set; }
```

Comment:
alternative names:
OptionallyDependsOn
OptionalDependsOn 
NonRequirements

Reply:
I was thinking about this exact thing a couple hours after making this PR, I like OptionallyDependsOn the best, and I've changed it in the PR.